pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -1107,6 +1174,46 @@ func (s *server) linearsearch(ctx context.Context, cfg *payload.Search_Config,
 			if num != 0 && len(res.GetResults()) > num {
 				res.Results = res.GetResults()[:num]
 			}
+
+			if errors.Is(ectx.Err(), context.DeadlineExceeded) {
+				if len(res.GetResults()) == 0 {
+					err = status.WrapWithDeadlineExceeded(
+						"error search result length is 0",
+						errors.ErrEmptySearchResult,
+						&errdetails.RequestInfo{
+							RequestId:   cfg.GetRequestId(),
+							ServingData: errdetails.Serialize(cfg),
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/vald.v1.search",
+							ResourceName: fmt.Sprintf("%s: %s(%s) to %v", apiName, s.name, s.ip, s.gateway.Addrs(ctx)),
+						}, info.Get(),
+					)
+					if span != nil {
+						span.SetStatus(trace.StatusCodeDeadlineExceeded(err.Error()))
+					}
+					return nil, err
+				}
+				if min != 0 && len(res.GetResults()) < min {
```

Comment:
the variable definition of `min` is `int`.
Is this condition right?

The below is more safety.
```suggestion
				if min >= 0 && len(res.GetResults()) < min {
```
Or, the definition of `min` changes to `uint`.

Reply:
Thank you. 
I think your suggestion is a good one if we consider more safety.
However, I have one concern. If the user doesn't set the min to SearchConfig, it defaults to 0, so I think `min>0` would be fine!

Reply:
I have fixed: https://github.com/vdaas/vald/pull/1546/commits/30c376e68b4d97ae157911cbb1d402c72c6b8c51